File: api/src/main/java/org/egglog/api/hospital/model/dto/response/HospitalAuthResponse.java
package org.egglog.api.hospital.model.dto.response;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import jakarta.persistence.*;
import lombok.*;
import org.egglog.api.hospital.model.entity.Hospital;
import org.egglog.api.user.model.entity.User;

import java.time.LocalDateTime;

/**
 * packageName    : org.egglog.api.hospital.model.dto.response
 * fileName       : HospitalAuthResponse
 * author         : 김형민
 * date           : 2024-04-26
 * description    :
 * ===========================================================
 * DATE              AUTHOR             NOTE
 * -----------------------------------------------------------
 * 2024-04-26        김형민       최초 생성
 */
@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class HospitalAuthResponse {
    private String empNo;
    private Boolean auth;
    private LocalDateTime authRequestTime;
    private LocalDateTime confirmTime;
}

File: client/data/src/main/java/com/org/egglog/data/auth/model/HospitalAuthResponse.kt
package com.org.egglog.data.auth.model

import androidx.room.Entity
import com.org.egglog.data.util.LocalDateTimeSerializer
import com.org.egglog.domain.auth.model.HospitalAuth
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Entity
@Serializable
data class HospitalAuthResponse(
    val empNo: String,
    val auth: Boolean,
    @Serializable(with = LocalDateTimeSerializer::class) val authRequestTime: LocalDateTime?,
    @Serializable(with = LocalDateTimeSerializer::class) val confirmTime: LocalDateTime?,
    val nurseCertificationImgUrl: String,
    val hospitalCertificationImgUrl: String,
    @Contextual val hospitalInfo: UserHospitalResponse?
)

fun HospitalAuthResponse.toDomainModel(): HospitalAuth {
        return HospitalAuth(
            empNo, auth, authRequestTime, confirmTime, nurseCertificationImgUrl, hospitalCertificationImgUrl, hospitalInfo?.toDomainModel()
        )
}